### DISCUSSION TITLE
NewTokenLandingPage redirect is a POST?

### DISCUSSION THREAD

**Discussion by LPmaverick** [OWASP/www-project-csrfguard#81] [Q&A] [answered]
Shouldn't this be a GET redirect?

**Comment by forgedhallpass** [comment, marked as answer]
Hello @LPmaverick,

It is recommended not to use `GET` if the data submitted is potentially sensitive, and I'd say CSRF tokens could be considered as such.

**Comment by LPmaverick** [comment]
But shouldn't everything about the original request be thrown out?  The redirect is to the configured endpoint with 0 parameters from the original request to ensure it is CSRF immune.

**Reply by forgedhallpass** [reply]
> But shouldn't everything about the original request be thrown out? 

And you are certain it is not happening that way?

Looking at the logic, it should work correctly. See:
https://github.com/OWASP/www-project-csrfguard/blob/f6bacb71d324cb016d9f4533d4194e3da1d3a187/csrfguard/src/main/java/org/owasp/csrfguard/CsrfGuard.java#L333

**Comment by LPmaverick** [comment]
That is happening.  So there will be no data sent in the redirect that we would need to worry about, correct?

Also, the redirect endpoint should contain no state-changing functionality and be completely CSRF immune, right?  Which would mean that this logic https://github.com/OWASP/www-project-csrfguard/blob/f6bacb71d324cb016d9f4533d4194e3da1d3a187/csrfguard/src/main/java/org/owasp/csrfguard/CsrfGuard.java#L351-L357 isn't needed.

**Reply by LPmaverick** [reply]
If the endpoint is protected in the application, then I see that as an issue within the application.

**Reply by forgedhallpass** [reply]
> So there will be no data sent in the redirect that we would need to worry about, correct?

There is a test application in the project exactly for the purpose of testing different configurations and scenarios. Please use it.

> Which would mean that this logic ... isn't needed

The logic above is needed, because you might decide to include CSRF protection for the new token landing page. It's totally up to you.

> If the endpoint is protected in the application, then I see that as an issue within the application.

I don't see what issue are you talking about. You control the endpoint, and you control whether you want it expect CSRF tokens or not. If you happen to find a problem, please create an issue with a detailed description, steps on how to reproduce it and the actual vs expected behavior.
